core/modules/url/router/beforeEach.ts

```diff
@@ -44,16 +34,16 @@ export async function beforeEach (to: Route, from: Route, next) {
           next(dynamicRoutes[0])
         } else {
           Logger.error('Route not found ' + routeData['name'], 'dispatcher')()
-          next('/page-not-found')
+          next(localizedRoute('/page-not-found', currentStoreView().storeCode))
```

Comment:
couldn't we just have a default value for `storeCode` param in `localizedRoute`? There links would look much prettier with just:
```suggestion
          next(localizedRoute('/page-not-found'))
```